lib/organize_letters.rb

```diff
@@ -0,0 +1,36 @@
+require_relative './ds/heap'
+
+def data = Struct.new(:value, :frequency)
+
+def organize_letters(string)
+  letters = Hash.new(0)
+  queue = Heap.new
+  string_size = string.size
+  new_string = ""
+  
+  string.each_char { |char| letters[char] += 1 }
+
+  letters.each do |letter, frequency|
+    queue.insert_smallest_to_highest(data.new(letter, frequency))
+  end
```

Comment:
Adds each pair value, frequency to the queue, value is the letter so we will have something like this in the example:

![image](https://github.com/ImMPrada/short-ds-challenges/assets/26731448/47bbf144-0964-4554-b80a-d3a92ee18ce7)